consensus/hotstuff/voteaggregator/vote_collectors.go

```diff
@@ -105,15 +109,18 @@ func (v *VoteCollectors) getCollector(view uint64) (hotstuff.VoteCollector, bool
 // kept and the method call is a NoOp.
 func (v *VoteCollectors) PruneUpToView(lowestRetainedView uint64) {
 	v.lock.Lock()
-	defer v.lock.Unlock()
```

Comment:
Was wondering what your motivation is to replace the `defer unlock` with explicit statements. It seems the code would be much easier to accidentally break with subsequent revisions.

Reply:
true, but I wanted to keep logging outside of critical section, maybe it's not worth it, but I always feel uncomfortable when locking more than needed.

Reply:
No strong feeling about it either. I wouldn’t worry too much about a log message extra within the scope of the lock … its not that we are computing much for this lock message. Either way is fine.